prov/efa/src/efa_hmem.c

```diff
@@ -78,6 +79,7 @@ static int efa_domain_hmem_info_init_protocol_thresholds(struct efa_domain *efa_
 	case FI_HMEM_CUDA:
 	case FI_HMEM_NEURON:
 		info->runt_size = EFA_DEFAULT_RUNT_SIZE;
+		info->max_gdrcopy_from_device_size = 0;
```

Comment:
> I do not think we have to maintain two thresholds

Are you referring to `max_gdrcopy_from_device_size` vs `shm_max_medium_size`, or something else?

Personally I think the current name makes more sense in:
```c
iface == FI_HMEM_CUDA &&
cuda_is_gdrcopy_enabled() &&
tx_entry->total_len <= hmem_info[FI_HMEM_SYSTEM].max_gdrcopy_from_device_size
```
which is more explicit about cuda and gdr.

A more generic name `max_copy_from_device_size`  obfuscates the intention. Meanwhile, so far I don't know the use case for a max limit for cudaMemcpy - it is the defacto method for copying large cuda messages AFAIK.

Please correct me if I was mistaken.

Reply:
I was referring to `max_gdrcopy_from_device_size` and `max_cudamemcopy_from_device_size` (if we care). I do not think it hurt to use a more generic name as long as we make sure to use the correct copy method in the protocol selection. `ofi_hmem_memcpy` as a wrapper does not distinguish cudamemcpy and gdrcopy in its name. So using a generic name for `copy_from_device` makes sense to me.

Reply:
The macro `EFA_DEFAULT_INTRA_MAX_GDRCOPY_FROM_DEV_SIZE` does not need a rename because it is the size for gdrcopy. The field in `info` should be generic. All these discrepancy are due to the shenanigans of Nvidia. We do not have to make our code comply with that

Reply:
> ofi_hmem_memcpy as a wrapper does not distinguish cudamemcpy and gdrcopy in its name

I see. It makes sense from ofi_hmem perspective. My perspective is to help the developer understand this threshold. IMO `max_gdrcopy_from_device_size` does better to remind the reader that this value is only relevant for gdrcopy, not cudaMemcpy. But I'm open to rename it once we have a use case for cudaMemcpy, and want to reuse this variable.

My intention for `info->max_gdrcopy_from_device_size = 0;` is to make it clear that gdrcopy does not support copying from cuda to this device(cuda or neuron) - does that make sense?

Reply:
Now it makes sense to me. Thanks.

Reply:
Thank you. Added a comment to clarify the intention.

Reply:
@shijin-aws I clarified with @wzamazon about the `efa_hmem_info` variable - the iface refers the user buffer type specifically. What we really want is a threshold for eager protocol based on the user buffer type.

Please see update.